### ISSUE TITLE
io.github.bonigarcia.wdm.WebDriverManagerException: java.io.FileNotFoundException: C:\Windows\TEMP\1120035346998766710\chromedriver.exe (Access is denied)

### ISSUE THREAD

**Issue by sandip05348** [bonigarcia/webdrivermanager#329]
Hi, 
I am facing given below issue. Need your valuable suggestion.
[INFO] Scanning for projects...
[INFO] 
[INFO] Using the builder org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder with a thread count of 1
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building WebDriverManagerDemo 0.0.1-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ WebDriverManagerDemo ---
[INFO] Deleting D:\MyworkSpace\WebDriverManagerDemo\target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ WebDriverManagerDemo ---
[WARNING] Using platform encoding (Cp1252 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 0 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.0:compile (default-compile) @ WebDriverManagerDemo ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ WebDriverManagerDemo ---
[WARNING] Using platform encoding (Cp1252 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] Copying 0 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.0:testCompile (default-testCompile) @ WebDriverManagerDemo ---
[INFO] Changes detected - recompiling the module!
[WARNING] File encoding has not been set, using platform encoding Cp1252, i.e. build is platform dependent!
[INFO] Compiling 1 source file to D:\MyworkSpace\WebDriverManagerDemo\target\test-classes
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.0:test (default-test) @ WebDriverManagerDemo ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running TestSuite
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
io.github.bonigarcia.wdm.WebDriverManagerException: java.lang.NullPointerException
	at io.github.bonigarcia.wdm.WebDriverManager.handleException(WebDriverManager.java:523)
	at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:471)
	at io.github.bonigarcia.wdm.WebDriverManager.handleException(WebDriverManager.java:520)
	at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:471)
	at io.github.bonigarcia.wdm.WebDriverManager.setup(WebDriverManager.java:229)
	at DriverPackage.MainDriver.beforeTest(MainDriver.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:59)
	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:458)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:222)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:142)
	at org.testng.TestRunner.beforeRun(TestRunner.java:529)
	at org.testng.TestRunner.run(TestRunner.java:497)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:383)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:344)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:125)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:417)
Caused by: java.lang.NullPointerException
	at io.github.bonigarcia.wdm.WebDriverManager.getDefaultBrowserVersion(WebDriverManager.java:958)
	at io.github.bonigarcia.wdm.ChromeDriverManager.getBrowserVersion(ChromeDriverManager.java:68)
	at io.github.bonigarcia.wdm.WebDriverManager.getVersionForInstalledBrowser(WebDriverManager.java:477)
	at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:430)
	... 32 more
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 3.311 s <<< FAILURE! - in TestSuite
[ERROR] f(DriverPackage.MainDriver)  Time elapsed: 0.029 s  <<< FAILURE!
java.lang.NullPointerException
	at DriverPackage.MainDriver.f(MainDriver.java:22)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   MainDriver.f:22 NullPointer
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 13.097 s
[INFO] Finished at: 2018-12-04T09:20:38+05:30
[INFO] Final Memory: 14M/33M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.0:test (default-test) on project WebDriverManagerDemo: There are test failures.
[ERROR] 
[ERROR] Please refer to D:\MyworkSpace\WebDriverManagerDemo\target\surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date]-jvmRun[N].dump, [date].dumpstream and [date]-jvmRun[N].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

**Comment by sandip05348** [comment]
public class Utility {

	public WebDriver driver;

	public Utility(WebDriver driver) {
		this.driver=driver;
		PageFactory.initElements(driver, this);
	}
	public HomePage startBrowser(String browserName) {
		if(browserName.equalsIgnoreCase("Chrome")){
			/*System.setProperty("webdriver.chrome.driver", System.getProperty("user.dir") + File.separator + "src" + File.separator 
					+ "main" + File.separator + "resources" + File.separator + "chromedriver.exe");*/
			ChromeOptions options = new ChromeOptions();
			options.setPageLoadStrategy(PageLoadStrategy.NONE);
			WebDriverManager.chromedriver().setup();
			driver= new ChromeDriver(options);
		}
		else if(browserName.equalsIgnoreCase("Firefox")){
			/*System.setProperty("webdriver.firefox.marionette", System.getProperty("user.dir") + File.separator + "src" + File.separator 
					+ "main" + File.separator + "resources" + File.separator + "geckodriver.exe");*/
			WebDriverManager.firefoxdriver().setup();
			driver= new FirefoxDriver();
		}
		else if(browserName.equalsIgnoreCase("IE")){
			WebDriverManager.iedriver().setup();
			 DesiredCapabilities ieCapabilities = DesiredCapabilities.internetExplorer();
			 ieCapabilities.setCapability(InternetExplorerDriver.INTRODUCE_FLAKINESS_BY_IGNORING_SECURITY_DOMAINS, true);
			 ieCapabilities.setCapability("ignoreZoomSetting", true);
			 ieCapabilities.setCapability("ignoreProtectedModeSettings" , true);
			driver= new InternetExplorerDriver(ieCapabilities);
		}
		else if(browserName.equalsIgnoreCase("Edge")){
			WebDriverManager.edgedriver().setup();
			driver= new EdgeDriver();
		}
		return PageFactory.initElements(driver, HomePage.class);	
	}
	
	public HomePage setUrl( String url) throws Exception{
		driver.get(url);
		return PageFactory.initElements(driver, HomePage.class);
	}

	public void setDriver(WebDriver webdriver) {
		driver=webdriver;
	}
	public WebDriver getDriver() {
		return driver;
	}

	public void QuitDriverU() {
		this.driver.quit();
	}
	
	public HomePage initDriver(WebDriver driver) throws Exception {
		return PageFactory.initElements(driver, HomePage.class);
	}

}

**Comment by bonigarcia** [comment]
No clue.

In any case, some hints for future issues:

- Use the latest version of WebDriverManager
- Use a logger (e.g. logback or other)
- Describe your problem in simple words, avoiding useless snippets
- Format the text of your issue (improve readability)

**Comment by fselcuk** [comment]
hello Boni Garcia
I have an exception

io.github.bonigarcia.wdm.config.WebDriverManagerException: java.io.FileNotFoundException: C:\Users\selcuk\.cache\selenium\chromedriver\win32\99.0.4844.51\chromedriver.exe (Access is denied)

after I updated my chrome to version 99.0.4844.51. But on my computer selenium cannot automate chrome browser anymore since the folder under win32\99.0.4844.51 is empty and cannot grab the chromedriver.exe automatically. I tried to copy and paste to that folder but never allows me to do so.
